pulp_python/app/models.py

```diff
@@ -1,12 +1,31 @@
+import json
+import os
+
+from collections import namedtuple
+from gettext import gettext as _
 from logging import getLogger
+from urllib.parse import urljoin, urlparse
 
+from django.contrib.postgres.fields import ArrayField
 from django.db import models
 
-from pulpcore.plugin.models import Content, Importer, Publisher
-from django.contrib.postgres.fields import ArrayField
+from pulpcore.plugin.models import (Artifact, Content, ContentArtifact, RemoteArtifact, Importer,
+                                    ProgressBar, Publisher, RepositoryContent, PublishedArtifact,
+                                    PublishedMetadata)
+
+from pulpcore.plugin.changeset import (
+    BatchIterator,
+    ChangeSet,
+    PendingArtifact,
+    PendingContent,
+    SizedIterable,
+)
+
 
 log = getLogger(__name__)
 
+Delta = namedtuple('Delta', ('additions', 'removals'))
```

Comment:
We ought to move this to core. This named tuple is part of the Plugin API, weird that each plugin has to write one.